Death to INOPs

J-Hoskin opened the pull request:
This merge merges over many of the features in my CJ4-Extended Repo, plus a few extra cockpit features.

mattnischan commented:
@J-Hoskin,

Definitely appreciate the work and contribution! I think as a general comment, we have tried to edit as few globally used files as possible, to make this mod as maximally compatible with any other aircraft. As this is structured at the moment, it touches a lot of files that are used in multiple planes.

I think before accepting, we would want those changes pulled up into the CJ4 cockpit file so that the changes are guaranteed to be isolated to a single aircraft, and not conflict with any other mods of those shared files.

J-Hoskin commented:
@mattnischan Okay, fair enough. I'll see what I can do.

mattnischan commented:
@J-Hoskin, feel free to leave this open as you work on it, if you like. The PR will automatically pick up changes you make in your branch in your fork as you go.

J-Hoskin commented:
@mattnischan I'll let you know when its ready.

mattnischan commented:
Sounds good!

J-Hoskin commented:
Okay, I have encapsulated the xml files into the CJ4 directory. I kept the same subtemplate architecture cause I think its best but it is separate to all other add-ons.

mattnischan commented on workingtitle-project-cj4.xml:
```diff
@@ -5,5 +5,6 @@
 		<Package>PackageDefinitions\workingtitle-vcockpits-instruments-cj4.xml</Package>
 		<Package>PackageDefinitions\workingtitle-vcockpits-instruments-airliners.xml</Package>
 		<Package>PackageDefinitions\workingtitle-aircraft-cj4.xml</Package>
+		<Package>PackageDefinitions\workingtitle-base-aircraft-common.xml</Package>
```
It looks like you pulled this package, but maybe left this line still here by accident.

J-Hoskin replied:
Ok, I'll render these points.

mattnischan reviewed (commented):
Looks good! I think my only additional general comment is on the folder structure: we haven't used `CJ4_Extended` anywhere else in the project, so that might be a bit confusing for devs, thinking maybe it's a different addon to these mods.

mattnischan commented:
@J-Hoskin One more question,

With the way this is structured, if Asobo fixes things in the core ModelDefs for other buttons and switches we haven't modified so far, will we receive those updates or does this pull all the definitions into our space?

J-Hoskin commented:
> @J-Hoskin One more question,
> 
> With the way this is structured, if Asobo fixes things in the core ModelDefs for other buttons and switches we haven't modified so far, will we receive those updates or does this pull all the definitions into our space?

The files I added only overwrite the buttons which use those new templates in the Cockpit.xml. So if Asobo push changes for aircon knob functionality, our functionality will replace those new changes. If they add functionality to say the the  ELT which isn't using our new templates, then those new changes will take affect.

J-Hoskin commented:
@mattnischan this is perfected.

cwburnett commented:
@J-Hoskin are you still working on this one, or are the sounds you're working on for aircon and things going to be in a separate PR?

J-Hoskin commented:
@cwburnett @mattnischan yeh its good to go. There is some small sound bugs where if you switch to exterior view and but to interior, the air fan sounds stop. But imo thats fine enough for now.

J-Hoskin commented:
**CHANGE LOG:**
- Added functionality to air and heat knob (Pilot, Copilot, and Cabin Fan, Cockpit and Cabin Temperature, Climate Control, Source)
- Added seatbelt announcement sound and seatbelt light functionality
- Added sounds to the air fans
